src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1460,6 +1448,7 @@ void PhaseIdealLoop::count_opaque_loop_nodes(Node* n, uint& init, uint& stride)
     }
   }
 }
+#endif // ASSERT
```

Comment:
Noticed that after JDK-8342043, `count_opaque_loop_nodes()` can also be made debug-build only.